backend/app/api/game.py

```diff
@@ -1,14 +1,165 @@
+            if j:  # If J is not none
+                if arr1[x][y] != arr2[x][y]:  # If the original old array does not contain the letter
+                    y2 = y
+
+                    if y2 < 14 and arr2[x][y2+1]:
+                        while arr2[x][y2-1]:  # It will find where the word starts
+                            y2 = y2-1
+                        while arr2[x][y2]:  # It will start adding the letters to the word, from where it starts
+                            if x < 15 and y2 < 15:
+                                word += arr2[x][y2]
+
+
+                            # For checking if the word is connected to another word
+                            if num_turn > 0:
+                                if 0 <= x < 14 and arr2[x + 1][y2] or arr2[x - 1][y2]:
+                                    check1 = check1 + 1
+
+                            y2 = y2 + 1
+                            if y2 == 15:
+                                break
+
+
+                    x2 = x
+                    if x2 < 14 and arr2[x2+1][y]:
+                        while arr2[x2-1][y]:
+                            x2 = x2-1
+                        while arr2[x2][y]:
+                            if x2 < 15 and y < 15:
+                                word += arr2[x2][y]
+
+                            # For checking if the word is connected to another word
+                            if num_turn > 0:
+                                if 0 <= y < 14 and arr2[x2][y + 1] or arr2[x2][y - 1]:
+                                    check2 = check2 + 1
+                            x2 = x2 + 1
+                            if x2 == 15:
+                                break
+
+
+            if len(word) == 0:
+                y += 1
+                if y == 15:
+                    y = 0
+            else:
+                break
+
+        if len(word) == 0:
+            x += 1
+            y = 0
+        else:
+            break
+
+
+    if len(word) == 0:
+        return ""
+    elif num_turn > 0 and check1 == 0 and check2 == 0:
+        return ""
+    else:
+        # For now it searches the JSON file
+        f = open('words_dictionary.json')
+
+        data = json.load(f)
+
+        for i in data:
+            if str(i).upper() == word:
+                return word
+
+    return ""
+
+
+
+# Checks if the start of the game is valid
+# The first letter must be placed in the middle of the board ([7][7])
+# The rest of the letters must be placed across or down from the center
+def valid_start(arr):
+    valid = False
+
+    if num_turn == 0 and arr[7][7]:
+        if arr[6][7] is None and arr[7][6] is None:
+            valid = True
+    elif num_turn > 0:
+        valid = True
+
+    return valid
+
+
+def calculate_Points(word):
+    total_points = 0
+
+    for letter in word:
+        total_points += points[letter]
+
+    return total_points
+
+
+# Gets a random tile from the tiles list
+def get_Tile():
+    index = random.randint(0, len(tiles))
```

Comment:
I used randint(0, len(tiles)) because I needed to remove the tile at the end by using its index. Is that still possible by using random.choice()?

Reply:
you can use `list.remove(item)` to remove a given item from a list. Honestly though it might be better to change the list to a dictionary instead and have the key as the letter and the value as the number of times it's allowed. Like 
`{'A': 2, 'B': 5}`.

I'm not saying you need to change this, just something to consider.

Reply:
I like the idea of using a dictionary, but wouldn't this give all the letters the same probability of being picked at the start. After a couple of turns, only the most popular letters will be left.